File: app/src/main/java/com/devdream/cookall/core/api/responses/RecipeListResponse.java
package com.devdream.cookall.core.api.responses;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class RecipeListResponse extends BaseResponse<List<RecipeListResponse>> {

    @SerializedName("id")
    private Long id;

    @SerializedName("title")
    private String title;

    @SerializedName("description")
    private String description;

    @SerializedName("cookingTime")
    private String cookingTime;

    @SerializedName("calories")
    private Float calories;

    @SerializedName("likes")
    private Long likes;

    @SerializedName("favorites")
    private Long favorites;

    @SerializedName("profileId")
    private Long profileId;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getCookingTime() {
        return cookingTime;
    }

    public void setCookingTime(String cookingTime) {
        this.cookingTime = cookingTime;
    }

    public Float getCalories() {
        return calories;
    }

    public void setCalories(Float calories) {
        this.calories = calories;
    }

    public Long getLikes() {
        return likes;
    }

    public void setLikes(Long likes) {
        this.likes = likes;
    }

    public Long getFavorites() {
        return favorites;
    }

    public void setFavorites(Long favorites) {
        this.favorites = favorites;
    }

    public Long getProfileId() {
        return profileId;
    }

    public void setProfileId(Long profileId) {
        this.profileId = profileId;
    }

}

File: app/src/main/java/com/devdream/cookall/core/api/responses/RecipeListResponse.kt
package com.devdream.cookall.core.api.responses

import com.google.gson.annotations.SerializedName

class RecipeListResponse : BaseResponse<List<RecipeListResponse>>() {

    @SerializedName("id")
    var id: Long? = null

    @SerializedName("title")
    var title: String? = null

    @SerializedName("description")
    var description: String? = null

    @SerializedName("cookingTime")
    var cookingTime: String? = null

    @SerializedName("calories")
    var calories: Float? = null

    @SerializedName("likes")
    var likes: Long? = null

    @SerializedName("favorites")
    var favorites: Long? = null

    @SerializedName("profileId")
    var profileId: Long? = null

}
